app/src/main/java/com/codesoom/assignment/application/AuthenticationService.java

```diff
@@ -0,0 +1,50 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.domain.User;
+import com.codesoom.assignment.domain.UserRepository;
+import com.codesoom.assignment.dto.UserLoginData;
+import com.codesoom.assignment.errors.PasswordNotCorrectException;
+import com.codesoom.assignment.errors.UserNotFoundException;
+import com.codesoom.assignment.utils.JwtUtil;
+import org.springframework.stereotype.Service;
+
+/**
+ * 인증 관련 비즈니스 로직을 처리합니다.
+ */
+@Service
+public class AuthenticationService {
+
+    private JwtUtil jwtUtil;
+
+    private UserRepository userRepository;
+
+    public AuthenticationService(JwtUtil jwtUtil, UserRepository userRepository) {
+        this.jwtUtil = jwtUtil;
+        this.userRepository = userRepository;
+    }
+
+    /**
+     * 사용자 로그인 정보로 로그인하고 토큰을 발행해 리턴합니다.
+     * @param userLoginData 로그인 하기 위한 사용자 정보
+     * @return 발행 한 토큰
+     */
+    public String login(UserLoginData userLoginData) {
+        User foundUser = userRepository.findByEmail(userLoginData.getEmail())
+                                       .orElseThrow(() -> new UserNotFoundException(userLoginData.getEmail()));
+
+        if (!foundUser.authenticate(userLoginData.getPassword())) {
+            throw new PasswordNotCorrectException();
+        }
+
+        return jwtUtil.encode(userLoginData.getEmail());
+    }
+
+    /**
+     * 엑세스 토큰을 디코드해서 사용자 이메일을 리턴합니다.
+     * @param accessToken 엑세스 토큰
+     * @return 사용자 이메일
+     */
+    public String parseToken(String accessToken) {
```

Comment:
설명도, 코드도 사용자 이메일을 리턴한다고 주장하고 있습니다. 그런데 메소드 시그니처가 JavaDoc과 코드와 괴리감이 있습니다. 따라서 몇 가지 선택 가능한 옵션이 있습니다.

첫번째 옵션은 다음과 같이 메소드 이름을 수정하는 것입니다.
```suggestion
     * 엑세스 토큰을 디코드해서 사용자 이메일을 리턴합니다.
     * @param accessToken 엑세스 토큰
     * @return 사용자 이메일
     */
    public String getUserEmailFromToken(String accessToken) {
```

두 번째는 리턴 타입을 변경하는 것입니다. (`Email` 타입을 만들어줘야 하겠죠.)

```suggestion
     * 엑세스 토큰을 디코드해서 사용자 이메일을 리턴합니다.
     * @param accessToken 엑세스 토큰
     * @return 사용자 이메일
     */
    public Email parseToken(String accessToken) {
```

세번째는 주석만 수정하는 것입니다.
```suggestion
     * 엑세스 토큰을 디코드해서 사용자 정보 문자열을 리턴합니다.
     * @param accessToken 엑세스 토큰
     * @return 사용자 이메일
     */
    public String parseToken(String accessToken) {
```

Reply:
작성하면서도 어색하단 느낌을 많이 받았었던 부분인데 제안해주신 3가지 다 괜찮네요.
JavaDoc을 작성하면서 메소드 시그니쳐의 중요성을 느끼게 되네요.